sql/core/src/main/scala/org/apache/spark/sql/streaming/DataStreamWriter.scala

```diff
@@ -290,8 +284,8 @@ final class DataStreamWriter[T] private[sql](ds: Dataset[T]) {
         df,
         dataSource.createSink(outputMode),
         outputMode,
-        useTempCheckpointLocation = useTempCheckpointLocation,
-        recoverFromCheckpointLocation = recoverFromCheckpointLocation,
+        useTempCheckpointLocation = true,
```

Comment:
Actually, I am implementing kafka source for kafka0.8. When I ran the tests for my implementation, I find the `ConsoleSink` cannot **continue** to process data.

So I think if we provide `checkPointLocation` for the **sinks**,  it can continue to process data. If we do not provide this option, it is ok  to start from beginning.  Not only the **other sinks**, I think all the sinks should follow the pattern above.